README.md

```diff
@@ -1 +1,145 @@
+
+## Instructions
+
+1. Push your solution to a **private repo** in your **personal Github account**
+2. When you are ready for us to take a look, please add [adjust-frontend-hiring][adjust-frontend-hiring] (GitHub user) as collaborator
+
+## General notes
+
+- The final solution must be a working React application.
+  - It's up to you how to bootstrap a new React application.
+- You can use any third-party library if you see a good fit for it.
+- Bringing in best practices such as tests, type safety, linting is appreciated.
+- It's not necessary to heavily comment the code, please only leave comments where you think they are necessary to understand the related code.
+- Aspects to be assessed:
+  - Code Style – meaningful naming, reasonable conciseness and consistency with applied code style conventions (i.e. sticking to either single or double quotes when declaring strings, but not mixing).
+  - Quality Assurance – the solution you upload should work as expected, conforming to the specifications of the tasks.
+
+Good luck and happy coding! 🚀
+
+---
+
+## Exercise
+
+### TASK #1 - JavaScript
+
+Implement `prepareData` higher order function, which takes an object of filter params `{year, customerName}`, and returns a function that processes a list of missions, by only showing the ones that were launched in `year` and carried a payload belonging to `customerName`.
+
+**Observations:**
+
+- Missions should appear in inverse chronological order (sort), with the exception of those that carried more payloads should appear first.
+- Payloads are carried in the second stage of a rocket and they can belong to multiple customers.
+- It doesn't matter to which `customerName` 'program' each payload belongs to as long as `customerName` is the customer.
+
+**Example:**
+
+Considering we have the list of missions from this [data fixture][data-fixtures], and the following filter params:
+
+```js
+{
+  year: 2018,
+  customerName: "NASA"
+}
+```
+
+The expected result should be:
+
+```js
+[
+  {
+    flight_number: 62,
+    mission_name: "Iridium NEXT Mission 6",
+    payloads_count: 2,
+  },
+  {
+    flight_number: 72,
+    mission_name: "CRS-16",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 64,
+    mission_name: "CRS-15",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 60,
+    mission_name: "TESS",
+    payloads_count: 1,
+  },
+  {
+    flight_number: 59,
+    mission_name: "CRS-14",
+    payloads_count: 1,
+  },
+];
+```
+
+---
+
+### TASK #2 - React & Hooks
+
+Implement `<RocketsList>` component with the following specifications:
+
+- The component receives `filerParams` object (with the shape described in [task #1][task-1]) as a prop.
+- The component obtains a list of 'missions' from a [custom hook][custom-hook], which fetches **the whole list of missions** from the [SpaceX API][spacex-api] and processes them with `prepareData` function (from [task #1][task-1]) and `filterParams` prop.
+  - The mission data should only be fetched once.
```

Comment:
I have removed the note about "if props change, component should not refetch the data". I thought it might confuse our candidates and push to overcomplicate the `useRockets` hook, as we don't really cover scenario with changing props in the hometask.